lib/msal-common/test/client/DeviceCodeClient.spec.ts

```diff
@@ -174,5 +174,22 @@ describe("DeviceCodeClient unit tests", async () => {
             const client = new DeviceCodeClient(config);
             await expect(client.acquireToken(request)).to.be.rejectedWith(`${ClientAuthErrorMessage.DeviceCodeExpired.desc}`);
         }).timeout(6000);
+
+        it("Throw device code expired exception if the timeout expires", async () => {
+            sinon.stub(DeviceCodeClient.prototype, <any>"executePostRequestToDeviceCodeEndpoint").resolves(DEVICE_CODE_RESPONSE);
+            const tokenRequestStub = sinon.stub(BaseClient.prototype, <any>"executePostToTokenEndpoint");
+
+            tokenRequestStub.onFirstCall().resolves(AUTHORIZATION_PENDING_RESPONSE);
+
+            let deviceCodeResponse = null;
+            const request: DeviceCodeRequest = {
+                scopes: [...TEST_CONFIG.DEFAULT_GRAPH_SCOPE, ...TEST_CONFIG.DEFAULT_SCOPES],
+                deviceCodeCallback: (response) => deviceCodeResponse = response,
+                timeout: DEVICE_CODE_RESPONSE.interval - 2, // Setting a timeout less the interval polling time to ensure the timeout has expired before the second poll
+            };
+
+            const client = new DeviceCodeClient(config);
+            await expect(client.acquireToken(request)).to.be.rejectedWith(`${ClientAuthErrorMessage.DeviceCodeExpired.desc}`);
```

Comment:
Might be better to reject with a different error (specific to the user request timeout being hit, instead of the default `expires_in` value).

Reply:
I agree a different error would communicate much clearer to our users.

Reply:
@pkanher617 please re-review, added a specific error for this timeout